### PR TITLE
Adds VS Code to workstation setup

### PR DESCRIPTION
**Pull request by bpina**

### CODE REVIEW

File: scripts/opt-in/vscode.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+ME=`whoami`
+VSCODE_PREFERENCES_SOURCE_FOLDER="${MY_DIR}/files/vscode"
+VSCODE_PREFERENCES_DESTINATION_FOLDER="/Users/$ME/Library/Application Support/Code/User"
+
+rm -rf /tmp/"Visual Studio Code.app"
+rm -rf /tmp/vscode.zip
+
+curl -o /tmp/vscode.zip -L https://go.microsoft.com/fwlink/?LinkID=620882
```
Comment (hjhart) [inline scripts/opt-in/vscode.sh]: We can use `brew cask install visual-studio-code` to install vscode. Was that not an option?

Comment (bpina) [reply scripts/opt-in/vscode.sh]: Nice. We can change it if that's a requirement, but I'm not personally a fan of using slower moving package managers when something is directly available from the source and doesn't require os specific install scripts.

Comment (hjhart) [reply scripts/opt-in/vscode.sh]: Cask is much easier to maintain (the open source community) and is constantly updated every time we run this script. It also handles the symlink for us.

What we lose in slowness from the package manager we gain in long-term maintainability, which is important for `workstation-setup`.

**Comment by hjhart** [comment]
Okay, added some documentation about a new script that allows you to run a single script (e.g. installing VSCode only, or adding extensions.).

File: files/vscode/extensions.txt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+misogi.ruby-rubocop
+ms-vscode.atom-keybindings
+ms-vscode.sublime-keybindings
+msjsdiag.debugger-for-chrome
+noku.rails-run-spec-vscode
+sianglim.slim
+vayan.haml
+xabikos.JasmineSnippets
+xabikos.JavaScriptSnippets
+xabikos.ReactSnippets
```
Comment (jkepps) [inline files/vscode/extensions.txt:27]: here's a few suggestions

```
waderyan.gitblame
rebornix.ruby
tyriar.sort-lines
```

Comment (bpina) [reply files/vscode/extensions.txt:27]: The git extension we have now is a bit more interactive and detailed, so the `gitblame` extension may be redundant. The other two are great!

Comment (jkepps) [reply files/vscode/extensions.txt:27]: ah, and one more
`vscode-icons-team.vscode-icons`

**Pull request merged**